presto-geospatial-toolkit/src/test/java/com/facebook/presto/geospatial/rtree/TestFlatbush.java

```diff
@@ -0,0 +1,281 @@
+        Rectangle hit1 = new Rectangle(-2, -2, -1, -2);
+        assertEquals(rtree.findIntersections(hit1), Arrays.asList(rect1));
+        // This should test missing at the root level
+        Rectangle totalMiss = new Rectangle(10, 10, 12, 12);
+        assertEquals(rtree.findIntersections(totalMiss).size(), 0);
+        // This should test missing at the leaf level
+        Rectangle marginalMiss = new Rectangle(0, 0, 0, 0);
+        assertEquals(rtree.findIntersections(marginalMiss).size(), 0);
+    }
+
+    @Test
+    public void testTwoLevelFlatbush()
+    {
+        // This is the smallest Rtree with height > 2
+        // Also test for NaN behavior
+        List<Rectangle> items = new ArrayList();
+        Rectangle rect0 = new Rectangle(1, 1, 1, 1);
+        items.add(rect0);
+        Rectangle rect1 = new Rectangle(-1, -1, -1, -1);
+        items.add(rect1);
+        Rectangle rect2 = new Rectangle(1, -1, 1, -1);
+        items.add(rect2);
+
+        items.sort(rectangleComparator);
+        Flatbush rtree = new Flatbush<Rectangle>(items.toArray(new Rectangle[] {}), 2);
+
+        List<Rectangle> allResults = rtree.findIntersections(everything);
+        allResults.sort(rectangleComparator);
+        assertEquals(allResults, items);
+
+        Rectangle hit0 = new Rectangle(1, 1, 1, 1);
+        assertEquals(rtree.findIntersections(hit0), Arrays.asList(rect0));
+        Rectangle hit1 = new Rectangle(-1, -1, -1, -1);
+        assertEquals(rtree.findIntersections(hit1), Arrays.asList(rect1));
+        Rectangle hit2 = new Rectangle(1, -1, 1, -1);
+        assertEquals(rtree.findIntersections(hit2), Arrays.asList(rect2));
+        // Test hitting across parent nodes
+        Rectangle hit12 = new Rectangle(-1, -1, 1, -1);
+        List<Rectangle> results12 = rtree.findIntersections(hit12);
+        results12.sort(rectangleComparator);
+        assertEquals(results12, Arrays.asList(rect1, rect2));
+
+        // This should test missing at the root level
+        Rectangle totalMiss = new Rectangle(10, 10, 12, 12);
+        assertEquals(rtree.findIntersections(totalMiss).size(), 0);
+        // This should test missing at the leaf level
+        Rectangle marginalMiss = new Rectangle(0, 0, 0, 0);
+        assertEquals(rtree.findIntersections(marginalMiss).size(), 0);
+    }
+
+    @Test
+    public void testOctagonQuery()
+    {
+        OGCGeometryWrapper octagonA = new OGCGeometryWrapper(POLYGON_A);
+        OGCGeometryWrapper octagonB = new OGCGeometryWrapper(POLYGON_B);
+
+        OGCGeometryWrapper pointX = new OGCGeometryWrapper(POINT_X);
+        OGCGeometryWrapper pointY = new OGCGeometryWrapper(POINT_Y);
+        OGCGeometryWrapper pointZ = new OGCGeometryWrapper(POINT_Z);
+        OGCGeometryWrapper pointW = new OGCGeometryWrapper(POINT_W);
+
+        Flatbush rtree = new Flatbush<OGCGeometryWrapper>(new OGCGeometryWrapper[] {
+                pointX, pointY, pointZ, pointW
```

Comment:
move this to the previous line:

`Flatbush<OGCGeometryWrapper> rtree = new Flatbush<>(new OGCGeometryWrapper[] {pointX, pointY, pointZ, pointW});`